File: CozyFocus/Models/FocusModels.swift
import Foundation
import SwiftUI
import SwiftData

enum Companion: String, CaseIterable, Identifiable, Codable {
    case redPanda, capybara, rabbit, puppy, cat, horse

    var id: String { rawValue }
    var name: String {
        switch self {
        case .redPanda: "Red Panda"
        case .capybara: "Capybara"
        case .rabbit: "Rabbit"
        case .puppy: "Puppy"
        case .cat: "Cat"
        case .horse: "Horse"
        }
    }
    var symbol: String {
        switch self {
        case .redPanda: "🦊"
        case .capybara: "🦫"
        case .rabbit: "🐇"
        case .puppy: "🐶"
        case .cat: "🐈"
        case .horse: "🐴"
        }
    }
    var accent: Color {
        switch self {
        case .redPanda: .orange
        case .capybara: .brown
        case .rabbit: .purple
        case .puppy: .yellow
        case .cat: .pink
        case .horse: .teal
        }
    }
}

enum Cosmetic: String, CaseIterable, Identifiable, Codable {
    case flowerCrown, sunHat, bandana, roundGlasses, bowTie

    var id: String { rawValue }
    var name: String {
        switch self {
        case .flowerCrown: "Flower crown"
        case .sunHat: "Sun hat"
        case .bandana: "Bandana"
        case .roundGlasses: "Round glasses"
        case .bowTie: "Bow tie"
        }
    }
    var mark: String {
        switch self {
        case .flowerCrown: "🌼"
        case .sunHat: "👒"
        case .bandana: "🧣"
        case .roundGlasses: "👓"
        case .bowTie: "🎀"
        }
    }
    var price: Int {
        switch self {
        case .flowerCrown: 15
        case .sunHat: 20
        case .bandana: 10
        case .roundGlasses: 25
        case .bowTie: 15
        }
    }
}

@Model
final class FocusSession {
    @Attribute(.unique) var id: UUID
    var completedAt: Date
    var duration: TimeInterval
    var companionRaw: String
    var coinsEarned: Int

    init(completedAt: Date = .now, duration: TimeInterval, companion: Companion, coinsEarned: Int = 5) {
        self.id = UUID()
        self.completedAt = completedAt
        self.duration = duration
        self.companionRaw = companion.rawValue
        self.coinsEarned = coinsEarned
    }
}

@Model
final class InventoryEntry {
    @Attribute(.unique) var cosmeticRaw: String
    var acquiredAt: Date

    init(cosmetic: Cosmetic, acquiredAt: Date = .now) {
        self.cosmeticRaw = cosmetic.rawValue
        self.acquiredAt = acquiredAt
    }
}

File: CozyFocusAndroid/app/src/main/java/com/example/cozyfocus/model/FocusModels.kt
package com.example.cozyfocus.model

import androidx.compose.ui.graphics.Color

enum class CompanionAnimal(
    val id: String,
    val displayName: String,
    val symbol: String,
    val accentColor: Color,
    val baseFrequencyHz: Double
) {
    RED_PANDA("redPanda", "Red Panda", "🦊", Color(0xFFFF9233), 780.0),
    CAPYBARA("capybara", "Capybara", "🦫", Color(0xFF8D5B34), 230.0),
    RABBIT("rabbit", "Rabbit", "🐇", Color(0xFFAB77C2), 620.0),
    PUPPY("puppy", "Puppy", "🐶", Color(0xFFE5A93C), 430.0),
    CAT("cat", "Cat", "🐈", Color(0xFFE960A2), 480.0),
    HORSE("horse", "Horse", "🐴", Color(0xFF2EA09A), 180.0);

    companion object {
        fun fromRaw(raw: String): CompanionAnimal {
            return entries.firstOrNull { it.id.equals(raw, ignoreCase = true) } ?: RED_PANDA
        }
    }
}

enum class Cosmetic(
    val id: String,
    val displayName: String,
    val mark: String,
    val price: Int
) {
    FLOWER_CROWN("flowerCrown", "Flower crown", "🌼", 15),
    SUN_HAT("sunHat", "Sun hat", "👒", 20),
    BANDANA("bandana", "Bandana", "🧣", 10),
    ROUND_GLASSES("roundGlasses", "Round glasses", "👓", 25),
    BOW_TIE("bowTie", "Bow tie", "🎀", 15);

    companion object {
        fun fromRaw(raw: String?): Cosmetic? {
            if (raw == null) return null
            return entries.firstOrNull { it.id.equals(raw, ignoreCase = true) }
        }
    }
}
